bitcoin/tx.c

```diff
@@ -180,7 +180,36 @@ static int elements_tx_add_fee_output(struct bitcoin_tx *tx)
 void bitcoin_tx_set_locktime(struct bitcoin_tx *tx, u32 locktime)
 {
 	tx->wtx->locktime = locktime;
-	tx->psbt->tx->locktime = locktime;
+	tx->psbt->fallback_locktime = locktime;
+	tx->psbt->has_fallback_locktime = true;
+}
+
+/* FIXME Stolen from psbt_append_input; export? */
+static struct wally_tx_input *wally_tx_input_from_outpoint_sequence(const struct bitcoin_outpoint *outpoint,
+			u32 sequence)
+{
+	struct wally_tx_input *tx_in;
+    if (chainparams->is_elements) {
+        if (wally_tx_elements_input_init_alloc(outpoint->txid.shad.sha.u.u8,
+                               sizeof(outpoint->txid.shad.sha.u.u8),
+                               outpoint->n,
+                               sequence, NULL, 0,
+                               NULL,
+                               NULL, 0,
+                               NULL, 0, NULL, 0,
+                               NULL, 0, NULL, 0,
+                               NULL, 0, NULL,
+                               &tx_in) != WALLY_OK)
+            abort();
+    } else {
+        if (wally_tx_input_init_alloc(outpoint->txid.shad.sha.u.u8,
+                          sizeof(outpoint->txid.shad.sha.u.u8),
+                          outpoint->n,
+                          sequence, NULL, 0, NULL,
+                          &tx_in) != WALLY_OK)
+            abort();
+    }
```

Comment:
I am not a big expert on this part of the code but it is possible abort with some more useful message? I think this case will never be true in the same case, but do you think that it will be useful to print some more information?